Sources/SwiftDocC/Model/Rendering/RenderSectionTranslator/DeclarationsSectionTranslator.swift

```diff
@@ -35,39 +42,124 @@ struct DeclarationsSectionTranslator: RenderSectionTranslator {
-                    assert(declarationFragments != nil, "Overloaded symbols must have declaration fragments.")
-                    guard let declarationFragments else { continue }
+                return postProcessTokens(translatedDeclaration)
+            }
 
-                    let declarationTokens = declarationFragments.map(translateFragment)
-                    otherDeclarations.append(
-                        .init(
-                            tokens: declarationTokens,
-                            identifier: overloadReference.absoluteString
-                        )
-                    )
+            typealias OverloadDeclaration = (
+                declaration: [SymbolGraph.Symbol.DeclarationFragments.Fragment],
+                reference: ResolvedTopicReference
+            )
+
+            func renderOtherDeclarationsTokens(
+                from overloadDeclarations: [OverloadDeclaration],
+                displayIndex: Int,
+                commonFragments: [SymbolGraph.Symbol.DeclarationFragments.Fragment]
+            ) -> DeclarationRenderSection.OtherDeclarations {
+                var otherDeclarations: [DeclarationRenderSection.OtherDeclarations.Declaration] = []
+
+                for overloadDeclaration in overloadDeclarations {
+                    let translatedDeclaration = translateDeclaration(
+                        overloadDeclaration.declaration,
+                        commonFragments: commonFragments)
+                    otherDeclarations.append(.init(
+                        tokens: translatedDeclaration,
+                        identifier: overloadDeclaration.reference.absoluteString))
 
                     // Add a topic reference to the overload
-                    renderNodeTranslator.collectedTopicReferences.append(overloadReference)
+                    renderNodeTranslator.collectedTopicReferences.append(
+                        overloadDeclaration.reference
+                    )
                 }
-                return .init(declarations: otherDeclarations, displayIndex: overloads.displayIndex)
+
+                return .init(declarations: otherDeclarations, displayIndex: displayIndex)
             }
 
-            var declarations = [DeclarationRenderSection]()
+            var declarations: [DeclarationRenderSection] = []
             for pair in declaration {
                 let (platforms, declaration) = pair
 
-                let renderedTokens = declaration.declarationFragments.map(translateFragment)
+                let renderedTokens: [DeclarationRenderSection.Token]
+                let otherDeclarations: DeclarationRenderSection.OtherDeclarations?
+
+                // If this symbol has overloads, render their declarations as well.
+                if let overloads = symbol.overloadsVariants[trait],
+                    let overloadDeclarations = symbol.overloadsVariants[trait]?.references
```

Comment:
I tried that at first but i got a compile error because you can't use an `if let` with a non-optional value 😭

Reply:
It might be worth extracting this entire expression. I find it a bit hard to read where the body of the if-statement starts because it's 20 lines below the `if` keyword.